apiserver/admin.go

```diff
@@ -389,10 +403,20 @@ func (a *admin) fillLoginDetails(result *authResult, lastConnection *time.Time)
 	// Send back user info if user
 	if result.userLogin {
 		userTag := a.root.entity.Tag().(names.UserTag)
-		var err error
-		result.userInfo, err = a.checkUserPermissions(userTag, result.controllerOnlyLogin)
-		if err != nil {
-			return errors.Trace(err)
+		if token := a.root.token; token != nil {
+			controllerAccess, _ := permissionFromToken(token, names.ControllerTagKind)
+			modelAccess, _ := permissionFromToken(token, names.ModelTagKind)
+			result.userInfo = &params.AuthUserInfo{
+				Identity:         userTag.String(),
```

Comment:
Would it be OK to move this struct into /pkg? So we can share the same struct?

Reply:
prefer not to because the all the juju over the wire params are kept together